packages/features/tasker/internal-tasker.ts

```diff
@@ -0,0 +1,38 @@
+import { Task } from "./repository";
+import { type Tasker, type TaskTypes } from "./tasker";
+import tasksMap from "./tasks";
+
+/**
+ * This is the default internal Tasker that uses the Task repository to create tasks.
+ * It doens't have any external dependencies and is suitable for most use cases.
+ * To use a different Tasker, you can create a new class that implements the Tasker interface.
+ * Then, you can use the TaskerFactory to select the new Tasker.
+ */
+export class InternalTasker implements Tasker {
+  async create(type: TaskTypes, payload: string): Promise<void> {
+    await Task.create(type, payload);
+  }
+  async processQueue(): Promise<void> {
+    const tasks = await Task.getNextBatch();
+    const tasksPromises = tasks.map(async (task) => {
+      const taskHandlerGetter = tasksMap[task.type as keyof typeof tasksMap];
+      if (!taskHandlerGetter) throw new Error(`Task handler not found for type ${task.type}`);
+      const taskHandler = await taskHandlerGetter();
+      return taskHandler(task.payload)
```

Comment:
Alright, here's my more in-depth thoughts (and can be done in the future). A Task should be represented by it's own entity, so rather than `Task` being a class of static types, it should represent only itself (with it's id), so all methods would mutate it's own state, eg:

```ts
return taskHandler(payload).then((task) => task.succeed);
```

This way, tasks can be retried, mutated, canceled, etc without static methods.
It's only a `nit` by itself but could lead to a nicer structure.

Reply:
I don't think I'm following your example here. I wanted to keep the class simple by providing a simple interface to call predefined methods. Static seemed like the natural way to go about it. Of course this can evolve if requirements change.